includes/Hooks/FooterHook.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LabkiPackManager\Hooks;
+
+use MediaWiki\Html\Html;
+use LabkiPackManager\Domain\Page;
+use LabkiPackManager\Services\LabkiPackRegistry;
+use LabkiPackManager\Services\LabkiPageRegistry;
+use LabkiPackManager\Services\LabkiRefRegistry;
+use LabkiPackManager\Services\LabkiRepoRegistry;
+use MediaWiki\MediaWikiServices;
+use MediaWiki\Title\Title;
+use MediaWiki\Skin\Skin;
+
+/**
+ * FooterHook
+ *
+ * Hook handler that appends metadata about imported pages to the footer so users
+ * can easily identify the originating pack, repository, and version details.
+ */
+final class FooterHook {
+    private const MAX_COMMIT_DISPLAY = 12;
+
+    /**
+     * Handler for the SkinAfterContent hook.
+     *
+     * @param string &$data HTML being appended after the page content
+     * @param Skin $skin Skin instance used to render the page
+     *
+     * @return bool Always true to continue hook processing
+     */
+    public static function onSkinAfterContent( string &$data, Skin $skin ): bool {
+        $title = $skin->getTitle();
+        /** @var LabkiPageRegistry $pageRegistry */
+        $pageRegistry = MediaWikiServices::getInstance()->getService( 'LabkiPageRegistry' );
+        
+        // This approach of pageID also works
+        // $pageId = $title ? $title->getArticleID() : 0;
+        // $page = $pageId ? $pageRegistry->getPageByWikiPageId( (int)$pageId ) : null;
+
+        $page = $pageRegistry->getPageByFinalTitle( mb_strtolower( $title->getPrefixedText() ) );
+        if ( !$page ) {
+            return true;
+        }
+        if ( !self::shouldDisplayFooter( $title, $skin ) ) {
+            return true;
+        }
+
+        $skin->getOutput()->addModuleStyles( 'ext.LabkiPackManager.importFooter' );
+
+        $footerHtml = self::buildFooterHtml( $page );
+        if ( $footerHtml === null ) {
+            return true;
+        }
+
+        $data .= $footerHtml;
+        return true;
+    }
+
+    /**
+     * Determine whether the footer should be shown for the current request.
+     *
+     * @param Title|null $title
+     * @param Skin $skin
+     *
+     * @return bool
+     */
+    private static function shouldDisplayFooter( ?Title $title, Skin $skin ): bool {
+        if ( !$title || !$title->canExist() ) {
+            return false;
+        }
```

Comment:
no need to do this. if we check for "is this page one of ours" then we can be sure that it can exist.